### PR TITLE
Add a build overload to Swift bindings for targets

### PR DESCRIPTION
**Pull request by hartbit**
This is necessary for the SwiftPM PR to use the llbuild library for building: https://github.com/apple/swift-package-manager/pull/1537

### CODE REVIEW

File: products/llbuildSwift/BuildSystemBindings.swift [hunk 1]
```diff
@@ -668,6 +668,17 @@ public final class BuildSystem {
         }
     }
 
+    /// Build a specific target.
+    ///
+    /// The client is responsible for ensuring only one build is ever executing concurrently.
+    ///
+    /// - parameter target: Name of the target to build.
+    /// - returns: True if the build was successful, false otherwise.
+    public func build(target: String) -> Bool {
```
Comment (aciidb0mb3r) [inline products/llbuildSwift/BuildSystemBindings.swift]: Maybe this should be flipped with the node variant?

```swift
func build(target: String? = nil) -> Bool
func build(node: String) -> Bool
```

Comment (hartbit) [reply products/llbuildSwift/BuildSystemBindings.swift]: Yep, sounds logical

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Comment by aciidb0mb3r** [comment]
@ddunbar @dmbryson ping

**Pull request merged**